src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -563,13 +445,44 @@ static uct_iface_internal_ops_t uct_cuda_ipc_iface_internal_ops = {
     .ep_is_connected       = uct_cuda_ipc_ep_is_connected
 };
 
+static uct_cuda_ctx_rsc_t * uct_cuda_ipc_ctx_rsc_create(uct_iface_h tl_iface)
+{
+    uct_cuda_ipc_iface_t *iface = ucs_derived_of(tl_iface, uct_cuda_ipc_iface_t);
+    uct_cuda_ipc_ctx_rsc_t *ctx_rsc;
+    unsigned i;
+
+    ctx_rsc = ucs_malloc(sizeof(*ctx_rsc), "uct_cuda_ipc_ctx_rsc_t");
+    if (ctx_rsc == NULL) {
+        ucs_error("failed to allocate cuda ipc context resource struct");
+        return NULL;
+    }
+
+    for (i = 0; i < iface->config.max_streams; i++) {
+        ctx_rsc->queue_desc[i].stream = NULL;
+        ucs_queue_head_init(&ctx_rsc->queue_desc[i].event_queue);
+    }
+
+    return &ctx_rsc->super;
+}
+
+static void
+uct_cuda_ipc_ctx_rsc_destroy(uct_iface_h tl_iface, uct_cuda_ctx_rsc_t *base)
```

Comment:
pls don't name variables just "base" because it can be different in different context
here, it can be "cuda_ctx_rsc"